models/demos/tg/llama3_70b/tests/test_llama_model_galaxy.py

```diff
@@ -0,0 +1,331 @@
+        all_pccs.append(extract_pcc_from_log(output_pcc))
+
+        kl_divs = scipy.stats.entropy(
+            torch.nn.functional.softmax(pytorch_out, dim=-1), torch.nn.functional.softmax(tt_out, dim=-1), axis=-1
+        )
+        logger.info(f"Mean KL Divergence: {kl_divs.mean()}")
+
+        # Write the code to check top-5 and top-1 accuracy. It should show the
+        # percentage where the top-1 prediction in pytorch was in the top-5
+        # predictions in tt.
+        reference_top1 = np.argmax(pytorch_out, axis=-1)
+        top1_acc = top_k_accuracy_score(reference_top1, tt_out, k=1, labels=np.arange(tt_out.shape[-1]))
+        top5_acc = top_k_accuracy_score(reference_top1, tt_out, k=5, labels=np.arange(tt_out.shape[-1]))
+
+        all_top1.append(top1_acc)
+        all_top5.append(top5_acc)
+
+        logger.info(f"Mean Top-1: {top1_acc}")
+        logger.info(f"Mean Top-5: {top5_acc}")
+
+        if does_pass:
+            logger.info(f"[start_pos={start_pos}] {llama_version} Model output Passed!")
+        else:
+            logger.warning(
+                f"[start_pos={start_pos}] {llama_version} Model output Failed! PCC value is lower than {pcc}"
+            )
+            all_tests_pass = False
+
+    logger.info(f"Average PCC over {len(all_pccs)} tokens: {sum(all_pccs) / len(all_pccs)}")
+    logger.info(f"Average Top-1 over {len(all_top1)} tokens: {sum(all_top1) / len(all_top1)}")
+    logger.info(f"Average Top-5 over {len(all_top5)} tokens: {sum(all_top5) / len(all_top5)}")
+    # Check kv cache
+    # PyTorch output --------------------------------------------------------------------
+    pytorch_layer_present = [
+        pytorch_model.model.layers[0]
+        .attention.cache_k.clone()
+        .permute(0, 2, 1, 3)[:batch, ...],  # [batch, n_kv_heads, seq, head_dim]
+        pytorch_model.model.layers[0]
+        .attention.cache_v.clone()
+        .permute(0, 2, 1, 3)[:batch, ...],  # [batch, n_kv_heads, seq, head_dim]
+    ]
+
+    tt_layer_present_all = [ttnn.from_device(lp) for lp in tt_model.layers[0].attention.layer_past]
+    tt_layer_present_all = [
+        ttnn.to_torch(
+            lp, mesh_composer=ConcatMesh2DToTensor(device_mesh, dims=(1, 0), cluster_shape=cluster_shape)
+        ).transpose(0, 1)[:batch, ...]
+        for lp in tt_layer_present_all
+    ]
+
+    cache_test_pass = check_kv_cache(
+        pytorch_layer_present,
+        tt_layer_present_all,
+        generation_start_pos,
+        generation_length,
+        seq_len,
+        model_config["LLM_MODE"] == "prefill",
+        pcc,
+    )
+    all_tests_pass = all_tests_pass and cache_test_pass
+    if all_tests_pass:
+        logger.info(f"{llama_version} output Passed!")
+    else:
+        gc.collect()
+        logger.warning(f"{llama_version} output Failed!")
+        assert all_tests_pass, f"PCC value is lower than {pcc} for some of the outputs. Check Warnings!"
+
+
+@pytest.mark.timeout(24000)
```

Comment:
do we need a 6 hour timeout?

Reply:
We should not this is legacy reason, just removed it in new commit @ttmchiou